samza-core/src/main/java/org/apache/samza/processor/SamzaContainerController.java

```diff
@@ -44,7 +44,7 @@
 public class SamzaContainerController {
   private static final Logger log = LoggerFactory.getLogger(SamzaContainerController.class);
 
-  private final ExecutorService executorService;
+  private ExecutorService executorService;
   private volatile SamzaContainer container;
```

Comment:
Is this class thread-safe? If not, the `volatile` modifier is probably redundant here? If it is, there are other variables like `containerFuture` that are not thread-safe yet. Would be nice to fix one or both.